app/src/main/java/io/homeassistant/companion/android/sensors/TrafficStatsManager.kt

```diff
@@ -0,0 +1,173 @@
+import java.math.RoundingMode
+import kotlin.math.absoluteValue
+
+class TrafficStatsManager : SensorManager {
+    companion object {
+        private const val TAG = "TrafficStats"
+        private const val GB = 1000000000
+
+        val rxBytesMobile = SensorManager.BasicSensor(
+            "mobile_rx_gb",
+            "sensor",
+            R.string.basic_sensor_name_mobile_rx_gb,
+            R.string.sensor_description_mobile_rx_gb,
+            unitOfMeasurement = "GB"
+        )
+        val txBytesMobile = SensorManager.BasicSensor(
+            "mobile_tx_gb",
+            "sensor",
+            R.string.basic_sensor_name_mobile_tx_gb,
+            R.string.sensor_description_mobile_tx_gb,
+            unitOfMeasurement = "GB"
+        )
+        val rxBytesTotal = SensorManager.BasicSensor(
+            "total_rx_gb",
+            "sensor",
+            R.string.basic_sensor_name_total_rx_gb,
+            R.string.sensor_description_total_rx_gb,
+            unitOfMeasurement = "GB"
+        )
+        val txBytesTotal = SensorManager.BasicSensor(
+            "total_tx_gb",
+            "sensor",
+            R.string.basic_sensor_name_total_tx_gb,
+            R.string.sensor_description_total_tx_gb,
+            unitOfMeasurement = "GB"
+        )
+        private var hasCellular = false
+    }
+
+    override val enabledByDefault: Boolean
+        get() = false
+    override val name: Int
+        get() = R.string.sensor_name_traffic_stats
+
+    override val availableSensors: List<SensorManager.BasicSensor>
+        get() = if (hasCellular) {
+            listOf(rxBytesMobile, txBytesMobile, rxBytesTotal, txBytesTotal)
+        } else listOf(rxBytesTotal, txBytesTotal)
+
+    override fun requiredPermissions(): Array<String> {
+        return emptyArray()
+    }
+
+    override fun hasSensor(context: Context): Boolean {
+        val cm: ConnectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val networkInfo = cm.allNetworks
+        var networkCapabilities: NetworkCapabilities
+        for (item in networkInfo) {
+            networkCapabilities = cm.getNetworkCapabilities(item)
+            if (!hasCellular)
+                hasCellular = networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
+        }
+        return true
+    }
+    override fun requestSensorUpdate(
+        context: Context
+    ) {
+        updateMobileRxBytes(context)
+        updateMobileTxBytes(context)
+        updateTotalRxBytes(context)
+        updateTotalTxBytes(context)
+    }
+
+    private fun updateMobileRxBytes(context: Context) {
+
+        if (!isEnabled(context, rxBytesMobile.id))
+            return
+
+        var mobileRx = 0f
+        val icon = "mdi:radio-tower"
+
+        try {
+            mobileRx = TrafficStats.getMobileRxBytes().toFloat()
+        } catch (e: Exception) {
+            Log.e(TAG, "Error getting the mobile rx bytes", e)
+        }

```

Comment:
If we have an issue getting the value we shouldn't continue to update the sensor otherwise you'll get random `0 GB` sent.